### PR TITLE
feat: Initial implementation of video benchmark

### PR DESCRIPTION
**Pull request by qh681248**
#892

### PR Type

- Feature

### Description

Extend the existing benchmarking to include video benchmarking. This is done by loading a video file, extracting its frames and then using coreset algorithms to select best frames. The frames are then compiled and saved into a gif file. This is another visual benchmark i.e. it is subjective which algorithm is performing better.

### How Has This Been Tested?

Existing tests pass as expected.

No new tests have been introduced.

### Checklist before requesting a review
- [x] I have made sure that my PR is not a duplicate.
- [x] My code follows the style guidelines of this project.
- [x] I have ensured my code is easy to understand, including docstrings and comments where necessary.
- [x] I have performed a self-review of my code.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I have updated CHANGELOG.md, if appropriate.

### CODE REVIEW

**Review by bk958178 — CHANGES_REQUESTED** [review]
Thanks @qh681248. Some minor questions and requested changes below

File: benchmark/mnist_benchmark.py [hunk 1]
```diff
@@ -474,7 +474,9 @@ def _get_stein_solver(_size: int) -> MapReduce:
             train_data_umap[idx]
         )
         stein_kernel = SteinKernel(kernel, score_function)
-        stein_solver = SteinThinning(coreset_size=_size, kernel=stein_kernel)
+        stein_solver = SteinThinning(
+            coreset_size=_size, kernel=stein_kernel, regularise=False
```
[the remark this answers is not in the document]
Comment (qh681248) [reply benchmark/mnist_benchmark.py:486]: I have rerun MNIST benchmark, and while I was rerunning it anyway, I also changes some of the hyperparameters for MNIST benchmark.

File: benchmark/pounce_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    base_dir = os.path.dirname(os.path.abspath(__file__))
+
+    # Ensure paths are absolute and output directory exists
+    in_path = Path(os.path.join(base_dir, in_path)).resolve()
+    out_dir = Path(os.path.join(base_dir, out_dir)).resolve()
+    out_dir.mkdir(parents=True, exist_ok=True)
+
+    # Load and preprocess video frames
+    _, *image_data = imageio.v2.mimread(in_path)
+    raw_data = np.asarray(image_data)
```
[the remark this answers is not in the document]
Comment (qh681248) [reply benchmark/pounce_benchmark.py:58]: Yes, it needs to be jnp.asarray for it to be used for coreset algorithm but it needs to be np.asarray for us to save frames later as gif, I have added 
 ` data = Data(jnp.array(umap_data))`

File: benchmark/pounce_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    reshaped_data = raw_data.reshape(raw_data.shape[0], -1)
+    print(type(reshaped_data))
+
+    umap_model = umap.UMAP(densmap=True, n_components=25)
```
[the remark this answers is not in the document]
Comment (qh681248) [reply benchmark/pounce_benchmark.py:62]: `n_components=25` was in the original pounce example script, it was chosen here for consistency

**Review by bk958178 — APPROVED** [review]
Thanks @qh681248. LGTM. Approving and merging

**Pull request merged**